pkgs/by-name/ec/echidna/package.nix

```diff
@@ -0,0 +1,115 @@
+{
+  lib,
+  stdenv,
+  haskellPackages,
+  fetchFromGitHub,
+  # build dependencies
+  slither-analyzer,
+}:
+
+haskellPackages.mkDerivation rec {
+  pname = "echidna";
+  version = "2.2.6";
+
+  src = fetchFromGitHub {
+    owner = "crytic";
+    repo = "echidna";
+    tag = "v${version}";
+    sha256 = "sha256-5nzis7MXOqs0bhx2jrEexjZYZI2qY6D0D7AWO+SPs+A=";
+  };
+
+  isExecutable = true;
+
+  libraryToolDepends = with haskellPackages; [
+    hpack
+  ];
+
+  executableHaskellDepends = with haskellPackages; [
+    # base dependencies
+    aeson
+    base
+    containers
+    directory
+    hevm
+    MonadRandom
+    mtl
+    text
+    # library dependencies
+    ansi-terminal
+    async
+    base16-bytestring
+    binary
+    brick
+    bytestring
+    data-bword
+    data-dword
+    deepseq
+    exceptions
+    extra
+    filepath
+    hashable
+    html-conduit
+    html-entities
+    http-conduit
+    ListLike
+    optics
+    optics-core
+    process
+    random
+    rosezipper
+    semver
+    signal
+    split
+    strip-ansi-escape
+    time
+    unliftio
+    utf8-string
+    vector
+    vty
+    vty-crossplatform
+    wai-extra
+    warp
+    word-wrap
+    xml-conduit
+    yaml
+    # executable dependencies
+    code-page
+    filepath
+    hashable
+    optparse-applicative
+    time
+    with-utf8
+  ];
+
+  # Note: there is also a runtime dependency of slither-analyzer. So, let's include it.
+  executableSystemDepends = [ slither-analyzer ];
+
+  preConfigure = ''
+    hpack
+  '';
+
+  # https://github.com/NixOS/nixpkgs/pull/304352
+  postInstall =
+    with haskellPackages;
+    lib.optionalString (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64) ''
+      remove-references-to -t ${warp.out} "$out/bin/echidna"
+      remove-references-to -t ${wreq.out} "$out/bin/echidna"
+    '';
+
+  shellHook = "hpack";
```

Comment:
This can probably be dropped. It would fail anyways since the source isn't necessarily available when entering `nix-shell -A echidna`.

Reply:
done.